command/healthcheck/pki_role_allows_glob_wildcards.go

```diff
@@ -0,0 +1,124 @@
+package healthcheck
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-secure-stdlib/parseutil"
+)
+
+type RoleAllowsGlobWildcards struct {
+	Enabled            bool
+	UnsupportedVersion bool
+
+	RoleEntryMap map[string]map[string]interface{}
+}
+
+func NewRoleAllowsGlobWildcardsCheck() Check {
+	return &RoleAllowsGlobWildcards{
+		RoleEntryMap: make(map[string]map[string]interface{}),
+	}
+}
+
+func (h *RoleAllowsGlobWildcards) Name() string {
+	return "role_allows_glob_wildcards"
+}
+
+func (h *RoleAllowsGlobWildcards) IsEnabled() bool {
+	return h.Enabled
+}
+
+func (h *RoleAllowsGlobWildcards) DefaultConfig() map[string]interface{} {
+	return map[string]interface{}{}
+}
+
+func (h *RoleAllowsGlobWildcards) LoadConfig(config map[string]interface{}) error {
+	enabled, err := parseutil.ParseBool(config["enabled"])
+	if err != nil {
+		return fmt.Errorf("error parsing %v.enabled: %w", h.Name(), err)
+	}
+	h.Enabled = enabled
+
+	return nil
+}
+
+func (h *RoleAllowsGlobWildcards) FetchResources(e *Executor) error {
+	exit, _, roles, err := pkiFetchRoles(e, func() {
+		h.UnsupportedVersion = true
+	})
+	if exit || err != nil {
+		return err
+	}
+
+	for _, role := range roles {
+		skip, _, entry, err := pkiFetchRole(e, role, func() {
+			h.UnsupportedVersion = true
+		})
+		if skip || err != nil || entry == nil {
+			if err != nil {
+				return err
+			}
+			continue
+		}
+
+		h.RoleEntryMap[role] = entry
+	}
+
+	return nil
+}
+
+func (h *RoleAllowsGlobWildcards) Evaluate(e *Executor) (results []*Result, err error) {
+	if h.UnsupportedVersion {
+		// Shouldn't happen; roles have been around forever.
+		ret := Result{
+			Status:   ResultInvalidVersion,
+			Endpoint: "/{{mount}}/roles",
+			Message:  "This health check requires Vault 1.11+ but an earlier version of Vault Server was contacted, preventing this health check from running.",
+		}
+		return []*Result{&ret}, nil
+	}
+
+	for role, entry := range h.RoleEntryMap {
+		allowsWildcard, present := entry["allow_wildcard_certificates"].(bool)
```

Comment:
We test for type checks here but no where else? Are we using this as a gate to validate we have the other keys so no need to test/validate the others?

Reply:
Ohhhhhhhhhhhhhhhhhhhhhh I'm stupid.

I was expecting `present` to mean it was in the map, not that it was safely type-cast to `bool`. D'oh!